src/platform/backends/shared/base_virtual_machine_factory.h

```diff
@@ -127,4 +124,19 @@ inline void multipass::BaseVirtualMachineFactory::require_suspend_support() cons
 {
 }
 
+inline void multipass::BaseVirtualMachineFactory::require_clone_support() const
+{
+    throw NotImplementedOnThisBackendException{"clone"};
+}
+
+inline multipass::VirtualMachine::UPtr multipass::BaseVirtualMachineFactory::clone_vm_impl(
```

Comment:
Indeed, we do not have the consistency. But based on ricard, this style is slightly better because the client code can just glance the class and capture the methods he can have instead of having the implementation in their sight.